extensions/multiply-divide/README.md

```diff
@@ -0,0 +1,50 @@
+# General design
+
+**Extension State: Under Development**  
+**Requires: Base, VWI, CP2.0**  
+**CPUID Bit: CP2.3**
+
+# Overview
+
+This extension adds multiplication, division, and remainder instructions. 
+
+# Added Instructions
+
+These instructions are in the expanded calculation opcode section of instructions.
+
+A `U` at the front of an instruction indicates that it treats its operands as unsigned.
+An `S` indicates that the operands are treated as signed.
+
+### Opcode table
+
+| `C CCCC CCCC` | NAME     | Operation                                  | Flags  | Comment |
+|---------------|----------|--------------------------------------------|--------|---------|
+| `0 0001 0000` | `UDIV`   | `A ← A / B`                                | None   | (1)     |
+| `0 0001 0001` | `SDIV`   | `A ← A / B`                                | None   | (1)     |
+| `0 0001 0010` | `UREM`   | `A ← A % B`                                | None   | (1) (2) |
+| `0 0001 0011` | `SREM`   | `A ← A % B`                                | None   | (1) (2) |
+| `0 0001 0100` | `UMUL`   | `A ← A × B`                                | `CV`   | (3) (5) |
+| `0 0001 0101` | `SMUL`   | `A ← A × B`                                | `CV`   | (3) (6) |                                          
+| `0 0001 0110` | `UMULH`  | `A ← high(A × B)`                          | None   | (4)     |
+| `0 0001 0111` | `SMULH`  | `A ← high(A × B)`                          | None   | (4)     |                                          
+
+1) Non-integral results of divisions are truncated towards zero. Therefore 10 / 3 is 3,
+    and -10 / 3 is -3. The result (quotient or remainder) is zero extended for the unsigned
+    operations, and sign extended for the signed operations.
+2) The remainder of a division is always smaller in magnitude than
+    the divisor, and has the sign of the dividend.
+3) The multiplication operands are treated as having the operation size (as from the `SS` bits).
+    The result will have the next size up. If it cannot be stored (for example, the operand size
+    is `word` and the system does not support [doubleword operations](../double-word-operations)),
+    then the truncation to the operation size is stored. The result of `UMUL` is always zero-extended,
+    and the result of `SMUL` is always sign-extended.
```

Comment:
Since we briefly mentioned this scenario earlier, what do we do when the A argument is memory? Do we always truncate to `SS`? Do we always store the full result? Going with the latter, the programmer would need to ensure that the memory address is properly aligned.

Reply:
My preferred solution to handling the twice as large output is to have fixed result registers (eg. R0 and R1)